src/ruby/app/models/keyboard.rb

```diff
@@ -682,7 +688,28 @@ def add_layer(name, map)
       col_index += 1
     end
     @keymaps[name] = new_map
-    @layer_names << name
+    @layer_names << name unless @layer_names.include?(name)
+  end
+
+  def get_layer(name, num)
+    if name
+      return @keymaps[name] if @keymaps[name]
+    end
+
+    if @layer_names.size>num
+      return @keymaps[@layer_names[num]]
+    end
+
+    return nil
+  end
+
+  def delete_mode_keys(layer_name)
+    @composite_keys.delete_if { |item| item[:layer]==layer_name }
+    @mode_keys.delete_if { |item| item[:layer]==layer_name }
```

Comment:
@yswallow 
Sorry but, steep check failed after I resolved some conflicts
https://github.com/picoruby/prk_firmware/runs/6086923043?check_suite_focus=true#step:5:18
Do you figure it out?

Reply:
It is caused by `@mode_keys` changed  from `Array` to `Hash`.

I'll solve it.

Reply:
I think it is fixed by changing

```ruby
@mode_keys.delete_if { |item| item[:layer]==layer_name }
```

into

```ruby
@mode_keys.each do |switch, config|
  config[:layers].delete(layer_name)
end
```

But I can't build it. it seems `picorbc`'s bug in output filename.

in `prk_firmware/build`,

```
$ ls

$ ../lib/picoruby/bin/picorbc -B models ../src/ruby/test/tmp/models.rb

$ ls
'_'$'\232\177''g'

$

```

output filename should be `models.c`.

Reply:
@yswallow I don't know what happened on your end though, there is a mistake of mine in resolving a conflict.
See-> https://github.com/picoruby/prk_firmware/commit/31222c4bc7d209dc5ae559cf53c0af1a820af824
I could build with this fix

Reply:
Ah sorry, my bad. Let me think again...

Reply:
I tested.

But it prints in CDC

```
No keymap.rb found.
Error: Can't assign VM-ID.
```

and hang-up.